app/controllers/nonsipp/declaration/SummaryController.scala

```diff
@@ -128,39 +134,40 @@ class SummaryController @Inject() (
       }
 
       entries(srn)
-        .map(entries => Ok(view(postSubmissionViewModel(entries), schemeDate(srn), false, lastSubmittedDateTime)))
+        .map(entries =>
+          Ok(view(viewModel(entries, declarationLink(srn)), schemeDate(srn), false, lastSubmittedDateTime))
+        )
         .value
         .map(_.merge)
+    }
   }
 
-  def onPageLoadPostSubmission(srn: Srn, startDate: String): Action[AnyContent] = identifyAndRequireData(srn).async {
+  def onPageLoad(srn: Srn, startDate: String): Action[AnyContent] = identifyAndRequireData(srn).async {
     implicit request =>
       (for {
         versions <- EitherT.right[Result](
           psrVersionsService
             .getVersions(request.schemeDetails.pstr, startDate, srn)
         )
-        submittedVersions = versions
-          .filter(_.reportStatus == SubmittedAndSuccessfullyProcessed)
-        lastSubmitted = submittedVersions
-          .maxByOption(_.compilationOrSubmissionDate)
-          .getOrElse(versions.maxBy(_.compilationOrSubmissionDate))
-        lastSubmittedDateTime = lastSubmitted.compilationOrSubmissionDate
+        lastVersion = versions
```

Comment:
I think we should go with the highest version number here and not the date of submission. In theory it should be the same, but some other logic in the service also sorts / searches by the highest version.